ec/ecresource/deploymentresource/delete.go

```diff
@@ -20,82 +20,84 @@ package deploymentresource
-// returned from the API.
-func deleteResource(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	const maxRetries = 3
-	var retries int
-	timeout := d.Timeout(schema.TimeoutDelete)
-	client := meta.(*api.API)
-
-	return diag.FromErr(resource.RetryContext(ctx, timeout, func() *resource.RetryError {
-		if _, err := deploymentapi.Shutdown(deploymentapi.ShutdownParams{
-			API: client, DeploymentID: d.Id(),
-		}); err != nil {
-			if alreadyDestroyed(err) {
-				d.SetId("")
-				return nil
-			}
-			return resource.NonRetryableError(multierror.NewPrefixed(
-				"failed shutting down the deployment", err,
-			))
-		}
+func (r *Resource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	if !r.ready(&resp.Diagnostics) {
+		return
+	}
 
-		if err := WaitForPlanCompletion(client, d.Id()); err != nil {
-			if shouldRetryShutdown(err, retries, maxRetries) {
-				retries++
-				return resource.RetryableError(err)
-			}
-			return resource.NonRetryableError(err)
-		}
+	var state deploymentv2.DeploymentTF
+
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
 
-		if err := handleTrafficFilterChange(d, client); err != nil {
-			return resource.NonRetryableError(err)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	//TODO retries
+
+	if _, err := deploymentapi.Shutdown(deploymentapi.ShutdownParams{
+		API: r.client, DeploymentID: state.Id.Value,
+	}); err != nil {
+		if alreadyDestroyed(err) {
+			return
 		}
+	}
 
-		// We don't particularly care if delete succeeds or not. It's better to
-		// remove it, but it might fail on ESS. For example, when user's aren't
-		// allowed to delete deployments, or on ECE when the cluster is "still
-		// being shutdown". Sumarizing, even if the call fails the deployment
-		// won't be there.
-		_, _ = deploymentapi.Delete(deploymentapi.DeleteParams{
-			API: client, DeploymentID: d.Id(),
-		})
-
-		d.SetId("")
-		return nil
-	}))
+	if err := WaitForPlanCompletion(r.client, state.Id.Value); err != nil {
+		resp.Diagnostics.AddError("deployment deletion error", err.Error())
+		return
+	}
+
+	// We don't particularly care if delete succeeds or not. It's better to
+	// remove it, but it might fail on ESS. For example, when user's aren't
+	// allowed to delete deployments, or on ECE when the cluster is "still
+	// being shutdown". Sumarizing, even if the call fails the deployment
+	// won't be there.
+	_, _ = deploymentapi.Delete(deploymentapi.DeleteParams{
+		API: r.client, DeploymentID: state.Id.Value,
+	})
 }
 
 func alreadyDestroyed(err error) bool {
 	var destroyed *deployments.ShutdownDeploymentNotFound
 	return errors.As(err, &destroyed)
 }
 
-func shouldRetryShutdown(err error, retries, maxRetries int) bool {
-	const timeout = "Timeout exceeded"
-	needsRetry := retries < maxRetries
+func removeRule(ruleID, deploymentID string, client *api.API) error {
```

Comment:
IMO this doesn't make sense here. It's not called when deleting the deployment resource, it should live with the update code.